common/viewer.cpp

```diff
@@ -1583,6 +1583,132 @@ namespace rs2
                 }
             }
 
+            //switch( stream_mv.profile.stream_type() )
+            {
+                static std::vector< std::pair< ImColor, bool > > colors =
+                {
+                    { ImColor( 1.f, 1.f, 1.f, 1.f ), false },  // the default color
+                    { ImColor( 1.f, 0.f, 0.f, 1.f ), false },
+                    { ImColor( 0.f, 1.f, 0.f, 1.f ), false },
+                    { ImColor( 0.f, 0.f, 1.f, 1.f ), false },
+                    { ImColor( 1.f, 0.f, 1.f, 1.f ), false },
+                    { ImColor( 1.f, 1.f, 0.f, 1.f ), false },
+                };
+                typedef size_t ColorIdx;
+                static std::map< size_t, ColorIdx > id2color;
+
+                auto get_color = [&]( object_in_frame const & object ) -> ImColor
+                {
+                    ColorIdx & color = id2color[object.id];
```

Comment:
id2color[object.id % colors.size()]; ?

Reply:
The idea here is that the object ID would be mapped to a color index. You don't normally want two IDs mapping to the same color, and I assumed we know nothing about the nature of the ID.